controllers/activemqartemis_controller_test.go

```diff
@@ -2567,6 +2567,64 @@ var _ = Describe("artemis controller", func() {
 			Eventually(checkCrdDeleted(crd.Name, namespace, &crd), timeout, interval).Should(BeTrue())
 		})
 	})
+
+	It("populateValidatedUser", func() {
+
+		ctx := context.Background()
+		crd := generateArtemisSpec(defaultNamespace)
+		crd.Spec.DeploymentPlan.ReadinessProbe = &corev1.Probe{
+			InitialDelaySeconds: 1,
+			PeriodSeconds:       5,
+		}
+		crd.Spec.DeploymentPlan.Size = 1
+		crd.Spec.BrokerProperties = []string{
+			"securityEnabled=true",
+			"rejectEmptyValidatedUser=true",
+			"populateValidatedUser=true",
+		}
+
+		if os.Getenv("USE_EXISTING_CLUSTER") == "true" {
+
+			By("Deploying a broker")
+			Expect(k8sClient.Create(ctx, &crd)).Should(Succeed())
+
+			By("Checking ready on SS")
+			Eventually(func(g Gomega) {
+				key := types.NamespacedName{Name: namer.CrToSS(crd.Name), Namespace: defaultNamespace}
+				sfsFound := &appsv1.StatefulSet{}
+
+				g.Expect(k8sClient.Get(ctx, key, sfsFound)).Should(Succeed())
+				g.Expect(sfsFound.Status.ReadyReplicas).Should(BeEquivalentTo(1))
+			}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+			By("Sending 1")
+
+			podWithOrdinal := namer.CrToSS(crd.Name) + "-0"
+			command := []string{"amq-broker/bin/artemis", "producer", "--user", "Jay", "--password", "activemq", "--url", "tcp://" + podWithOrdinal + ":61616", "--message-count", "1", "--destination", "queue://DLQ", "--verbose"}
+
+			Eventually(func(g Gomega) {
+				stdOutContent := execOnPod(podWithOrdinal, crd.Name, defaultNamespace, command, g)
+				g.Expect(stdOutContent).Should(ContainSubstring("Produced: 1 messages"))
+			}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+			By("Consuming 1")
+
+			command = []string{"amq-broker/bin/artemis", "browser", "--user", "Jay", "--password", "activemq", "--url", "tcp://" + podWithOrdinal + ":61616", "--destination", "queue://DLQ", "--verbose"}
+
+			Eventually(func(g Gomega) {
+				stdOutContent := execOnPod(podWithOrdinal, crd.Name, defaultNamespace, command, g)
+
+				// ...ActiveMQMessage[ID:3a356c2a-e7e1-11ec-ae1c-5ec4168c91b2]:PERSISTENT/ClientMessageImpl[messageID=19, durable=true, address=DLQ,userID=3a356c2a-e7e1-11ec-ae1c-5ec4168c91b2,properties=TypedProperties[__AMQ_CID=3a2f9fc7-e7e1-11ec-ae1c-5ec4168c91b2,_AMQ_ROUTING_TYPE=1,_AMQ_VALIDATED_USER=73ykuMrb,count=0,ThreadSent=Producer ActiveMQQueue[DLQ], thread=0]]
```

Comment:
question?
what is odd is that there is some random string in the user field rather than the user name Jay that the producer uses. What can be the reason for that. 
Note: _AMQ_VALIDATED_USER=73ykuMrb
ideally I would assert the presence of _AMQ_VALIDATED_USER=Jay

Reply:
with the config as it is, there is a guest login module configured with an auto generated user name for the guest name, that is what is being propagated